rust/datafusion/src/physical_plan/functions.rs

```diff
@@ -280,6 +309,18 @@ fn signature(fun: &BuiltinScalarFunction) -> Signature {
             Signature::Uniform(1, vec![DataType::Utf8, DataType::LargeUtf8])
         }
         BuiltinScalarFunction::Concat => Signature::Variadic(vec![DataType::Utf8]),
+        BuiltinScalarFunction::CharacterLength => {
+            Signature::Uniform(1, vec![DataType::Utf8, DataType::LargeUtf8])
```

Comment:
The signature states that `LargeUtf8` is supported, but the implementation only supports `Utf8`.

If we only use `DataType::Utf8` here, the planner will coerce any `LargeUtf8` to `Utf8`. :)

Reply:
I have reworked the macros significantly to now support Utf8 and LargeUtf8 functions going forward. It would be trivial to add more functions like `ltrim` which support both types.